### ISSUE TITLE
🐛 Bug Report: Scaffolder's array of objects field not working

### ISSUE THREAD

**Issue by josecosta-shell** [backstage/backstage#20233]
### 📜 Description

I am testing all the available documented fields and came across this issue on [this one](https://github.com/backstage/backstage/blob/master/docs/features/software-templates/input-examples.md#array-with-another-types). 

### 👍 Expected behavior

It should display the elements properly.

### 👎 Actual Behavior with Screenshots

Nothing is displayed on the page besides the title. 
<img width="456" alt="image" src="https://github.com/backstage/backstage/assets/129780729/c7b48e93-d945-427d-9f45-3c9b41ed64ce">

### 👟 Reproduction steps

SImply follow [this](https://github.com/backstage/backstage/blob/master/docs/features/software-templates/input-examples.md#array-with-another-types). example and it should break.

### 📃 Provide the context for the Bug.

I am on Backstage ` "version": "1.18.0"` and `"@backstage/plugin-scaffolder-backend": "^1.17.0",` with `"@backstage/plugin-scaffolder-node": "^0.2.4"`

### 🖥️ Your Environment

This is happening running on localhost using in memory db.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

**Comment by Teemukoivumaa** [comment]
Because arrayObjects is type of array, it has by default zero items. You can test this by changing the `ui:options: addable` to be true and then you can add the items. Fix for this is to just add `minItems: 1` so it initialises with one item

```diff
parameters:
  - title: Fill in some steps
    properties:
      arrayObjects:
        title: Array with custom objects
        type: array
+       minItems: 1
        ui:options:
          addable: false
          orderable: false
          removable: false
        items:
          type: object
          properties:
            array:
              title: Array string with default value
              type: string
              default: value3
              enum:
                - value1
                - value2
                - value3
            flag:
              title: Boolean flag
              type: boolean
              ui:widget: radio
            someInput:
              title: Simple text input
              type: string
```

I can make a PR with updated docs 😃

**Comment by Teemukoivumaa** [comment]
PR: #20450

**Comment by Teemukoivumaa** [comment]
Oh btw the PR was merged 3 weeks ago

**Comment by vinzscam** [comment]
Closing the issue as fixed by #20450
